pkg/kubelet/dockershim/libdocker/kube_docker_client.go

```diff
@@ -459,7 +459,7 @@ func (d *kubeDockerClient) CreateExec(id string, opts dockertypes.ExecConfig) (*
 	return &resp, nil
 }
 
-func (d *kubeDockerClient) StartExec(startExec string, opts dockertypes.ExecStartCheck, sopts StreamOptions) error {
+func (d *kubeDockerClient) StartExec(startExec string, opts dockertypes.ExecStartCheck, sopts StreamOptions, eopts ExecOptions) error {
 	ctx, cancel := d.getCancelableContext()
```

Comment:
Would it make sense to set the timeout in the context?

Reply:
This patch allows the code that calls `StartExec` to be notified on timeout. Otherwise, each caller should have its own implementation for detecting timeouts.